src/libsystemd/sd-journal/journal-file.c

```diff
@@ -3447,6 +3447,8 @@ int journal_file_open(
 
         if (f->last_stat.st_size == 0 && f->writable) {
 
+                newly_created = true;
```

Comment:
this improves the situation a bit, but there are various earlier causes that might fail, hence is not a comprehensive solution. in particular mmap_cache_add_fd() is something that can fail. I think it would be best if the open() call would directly tell us whether it created a new file or not.

I prepped some code for this now → #22808

Please review, and then rebase this one

Reply:
yes agreed.